### PR TITLE
Infisical plugin and docs cleanup

### PR DESCRIPTION
**Pull request by philmillman**
Adds support for reading secrets from [Infisical](https://infisical.com/). This first iteration supports universal auth and fetching individual secrets by name.

We may want to break out some of the new content that was added to Overview and give it its own nav items.

### CODE REVIEW

**Comment by philmillman** [comment]
Implements https://github.com/dmno-dev/dmno/issues/119

File: packages/docs-site/src/content/docs/docs/plugins/1password.mdx [hunk 1]
```diff
@@ -242,16 +226,9 @@ Field IDs are not easy to get from the 1Password UI. Luckily when the supplied f
 The secret reference for invidivual fields within an item can be found by clicking on the down arrow icon **on the field** and selecting `Copy Secret Reference`.
 :::
 
-## Caching
```
Comment (theoephraim) [inline packages/docs-site/src/content/docs/docs/plugins/1password.mdx:245]: I think we should still mention this stuff, but just not go into the details.

File: packages/docs-site/src/content/docs/docs/guides/secret-segmentation.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+In general, you should apply the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) when it comes to secrets. The facets of this are different for each organization, but as a rule of thumb, you should aim to:
+- Minimize the number of people who have access to secrets, and regularly review and audit who does
+- Minimize the environments (e.g., dev, staging, production) and their associated machine identities that have access to secrets
+- Rotate secrets whenever possible
+- Use short-lived credentials whenever possible
+
+## Sample setups
+
+Asuming something resembling a trunk-based workflow (i.e., you have a single production environment and are using feature flags to manage changes across environments), the minimum viable setup that we recommend looks something like this:
```
Comment (theoephraim) [inline packages/docs-site/src/content/docs/docs/guides/secret-segmentation.mdx:16]:
```suggestion
Assuming something resembling a trunk-based workflow (i.e., you have a single production environment and are using feature flags to manage changes across environments), the minimum viable setup that we recommend looks something like this: 
```

File: packages/docs-site/src/content/docs/docs/plugins/bitwarden.mdx [hunk 3]
```diff
@@ -64,34 +54,12 @@ const bitwardenPlugin = BitwardenSecretsManagerDmnoPlugin.injectInstance('bitwar
...
 :::tip[Use projects to segment access]
-You should use multiple projects to segment your secrets following the [Principle of Least Privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). This means at minimum, you should have one project for ultra-sensitive production secrets, and another for everything else. How much more you want to break things up is up to you and will depend on your security requirements.
-
-It is technically possible to create secrets without using projects and/or to assign permissions at the individual secret level, but we do not recommend it.
+You should use multiple projects to segment your secrets following the [Principle of Least Privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege). See more in our [Secret segmentation](/docs/guides/secret-segmentation/) guide.
 :::
 
-## Setup Machine Account & Access Tokens
-
-Machine accounts can be granted access to projects, and each machine account can have multiple access tokens with optional expiration. How you want to manage this is up to you, but a sensible approach could be:
```
Comment (theoephraim) [inline packages/docs-site/src/content/docs/docs/plugins/bitwarden.mdx:74]: this is all specific to bitwarden's auth model, so might be good to leave in? It is also still part of the setup process, so at least still need to mention setting them up, and can then defer to the segmentation guide more about the strategy?

File: packages/docs-site/src/content/docs/docs/plugins/overview.mdx [hunk 4]
```diff
@@ -12,6 +12,96 @@ Our first set of plugins are designed to help you manage secrets:
...
+
+### Multiple plugin instances
+
+Plugins support multiple instances allowing you to compose your configuration in a flexible way. For example, with the 1Password plugin you could create separate instances for your production and non-production vaults. Each instance can have its own settings, allowing you to manage per-environment or per-service secrets.
+
+When defining a plugin instance, you provide an `id` which will be used to identify the instance in various places, such as the CLI, UI, and when injecting it into other services.
+
+### Injecting plugin instances in monorepo services
+
+In a monorepo, you might want to use the same plugin instance across multiple services. If you will be using the same settings for each service, you can initialize a plugin instance once in a parent service as seen above, and then inject it in child services. Note that the injected plugin instance must use the same id we set during initialization.
```
Comment (theoephraim) [inline packages/docs-site/src/content/docs/docs/plugins/overview.mdx:77]: you could mention that it is injects the pre-configured instance, avoiding the need to pull in the config and wire it up in each service

File: packages/plugins/infisical/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      extends: InfisicalTypes.clientId,
+    },
+    INFISICAL_CLIENT_SECRET: {
+      extends: InfisicalTypes.clientSecret,
+    },
+    INFISICAL_PROJECT_ID: {
+      extends: InfisicalTypes.projectId,
+    },
+    ITEM_FROM_INFISICAL: {
+      value: infisicalPlugin.secret()
```
Comment (theoephraim) [inline packages/plugins/infisical/README.md:44]: might want to mention the fetches it using the name, and then show an example with renaming?

File: packages/plugins/infisical/package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
+{
+  "name": "@dmno/infisical-plugin",
+  "version": "0.0.1",
```
Comment (theoephraim) [inline packages/plugins/infisical/package.json:3]: you can leave this at 0.0.0 and then changesets will bump it

**Review by theoephraim — APPROVED** [review]

File: packages/docs-site/src/content/docs/docs/plugins/overview.mdx (hunk 4 above)
Comment (philmillman) [inline packages/docs-site/src/content/docs/docs/plugins/overview.mdx:77]:
```suggestion
In a monorepo, you might want to use the same plugin instance across multiple services. If you will be using the same settings for each service, you can initialize a plugin instance once in a parent service as seen above, and then inject it in child services. This alleviates the need to have the necessary config in each additional service. Note that the injected plugin instance must use the same id we set during initialization.
```

File: packages/plugins/infisical/README.md (hunk 5 above)
Comment (philmillman) [reply packages/plugins/infisical/README.md:44]: yeah I'll update it to be more inline with the docs, I just noticed another error above

File: packages/plugins/infisical/package.json (hunk 6 above)
Comment (philmillman) [inline packages/plugins/infisical/package.json:3]:
```suggestion
  "version": "0.0.0",
```

File: packages/docs-site/src/content/docs/docs/plugins/bitwarden.mdx (hunk 3 above)
Comment (philmillman) [reply packages/docs-site/src/content/docs/docs/plugins/bitwarden.mdx:74]: ok sorry, if you want add the bitwarden specific stuff that makes sense back in, go for it, or just revert the whole then on this file

Comment (theoephraim) [reply packages/docs-site/src/content/docs/docs/plugins/bitwarden.mdx:74]: Leave it out for now and I’ll go back in and try to slim it down and link back to the guide

**Pull request merged**